### ISSUE TITLE
hyperv: Cluster not starting up / connection to port 8443 refused

### ISSUE THREAD

**Issue by bweissman** [kubernetes/minikube#3272]
min**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT 

Please provide the following details:

**Environment**:

**Minikube version** (use `minikube version`): v0.30.0
- **OS** (e.g. from /etc/os-release): Windows Server 2016 Datacenter on Azure VM
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): Hyper-V
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): v0.30.0
- **Install tools**:
- **Others**: 

**What happened**:
Process/Cluster doesn't come up:

Setting up kubeconfig...
Starting cluster components...
E1025 06:55:53.159797    5096 start.go:297] Error starting cluster:  kubeadm init error
sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
sudo /usr/bin/kubeadm alpha phase addon kube-dns
 running command: : running command:
sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI  &&
sudo /usr/bin/kubeadm alpha phase addon kube-dns

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):
minikube start --vm-driver="hyperv" --cpus 8 --memory 28672 --disk-size 100g --hyperv-virtual-switch "External"

**Output of `minikube logs` (if applicable)**:

tail:

Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.008497    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:460: Failed to list *v1.Node: Get https://localhost:8443/api/v1/nodes?fieldSelector=metadata.name%3Dminikube&limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.009399    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://localhost:8443/api/v1/pods?fieldSelector=spec.nodeName%3Dminikube&limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.010676    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:451: Failed to list *v1.Service: Get https://localhost:8443/api/v1/services?limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:31 minikube kubelet[3045]: I1025 08:22:31.276404    3045 kubelet_node_status.go:271] Setting node annotation to enable volume controller attach/detach
Oct 25 08:22:31 minikube kubelet[3045]: W1025 08:22:31.276858    3045 kubelet_node_status.go:1054] Failed to set some node status fields: can't get ip address of node minikube. error: No default routes.
Oct 25 08:22:31 minikube kubelet[3045]: W1025 08:22:31.276865    3045 kubelet_node_status.go:1054] Failed to set some node status fields: can't get ip address of node minikube. error: No default routes.
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.581943    3045 dns.go:131] Nameserver limits were exceeded, some nameservers have been omitted, the applied nameserver line is: 8.8.8.8 8.8.4.4 2001:4860:4860::8888
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.581943    3045 dns.go:131] Nameserver limits were exceeded, some nameservers have been omitted, the applied nameserver line is: 8.8.8.8 8.8.4.4 2001:4860:4860::8888
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.587197    3045 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.587239    3045 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "kube-apiserver-minikube_kube-system(f4b05ecfb3bc2c80b4821065f19bb199)" failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.587255    3045 kuberuntime_manager.go:646] createPodSandbox for pod "kube-apiserver-minikube_kube-system(f4b05ecfb3bc2c80b4821065f19bb199)" failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.587339    3045 pod_workers.go:186] Error syncing pod f4b05ecfb3bc2c80b4821065f19bb199 ("kube-apiserver-minikube_kube-system(f4b05ecfb3bc2c80b4821065f19bb199)"), skipping: failed to "CreatePodSandbox" for "kube-apiserver-minikube_kube-system(f4b05ecfb3bc2c80b4821065f19bb199)" with CreatePodSandboxError: "CreatePodSandbox for pod \"kube-apiserver-minikube_kube-system(f4b05ecfb3bc2c80b4821065f19bb199)\" failed: rpc error: code = Unknown desc = failed pulling image \"k8s.gcr.io/pause-amd64:3.1\": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable"
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.588988    3045 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.589098    3045 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "etcd-minikube_kube-system(bff6554ada19fd6d115435f7299bdd8a)" failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.589127    3045 kuberuntime_manager.go:646] createPodSandbox for pod "etcd-minikube_kube-system(bff6554ada19fd6d115435f7299bdd8a)" failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.589240    3045 pod_workers.go:186] Error syncing pod bff6554ada19fd6d115435f7299bdd8a ("etcd-minikube_kube-system(bff6554ada19fd6d115435f7299bdd8a)"), skipping: failed to "CreatePodSandbox" for "etcd-minikube_kube-system(bff6554ada19fd6d115435f7299bdd8a)" with CreatePodSandboxError: "CreatePodSandbox for pod \"etcd-minikube_kube-system(bff6554ada19fd6d115435f7299bdd8a)\" failed: rpc error: code = Unknown desc = failed pulling image \"k8s.gcr.io/pause-amd64:3.1\": Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable"
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.973261    3045 event.go:209] Unable to write event: 'Patch https://localhost:8443/api/v1/namespaces/default/events/minikube.1560c5fbb2553af4: dial tcp 127.0.0.1:8443: getsockopt: connection refused' (may retry after sleeping)
Oct 25 08:22:32 minikube kubelet[3045]: E1025 08:22:32.010135    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:460: Failed to list *v1.Node: Get https://localhost:8443/api/v1/nodes?fieldSelector=metadata.name%3Dminikube&limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:32 minikube kubelet[3045]: E1025 08:22:32.010696    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://localhost:8443/api/v1/pods?fieldSelector=spec.nodeName%3Dminikube&limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:32 minikube kubelet[3045]: E1025 08:22:32.012273    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:451: Failed to list *v1.Service: Get https://localhost:8443/api/v1/services?limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:33 minikube kubelet[3045]: E1025 08:22:33.011692    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:460: Failed to list *v1.Node: Get https://localhost:8443/api/v1/nodes?fieldSelector=metadata.name%3Dminikube&limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:33 minikube kubelet[3045]: E1025 08:22:33.012554    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://localhost:8443/api/v1/pods?fieldSelector=spec.nodeName%3Dminikube&limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused
Oct 25 08:22:33 minikube kubelet[3045]: E1025 08:22:33.013413    3045 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:451: Failed to list *v1.Service: Get https://localhost:8443/api/v1/services?limit=500&resourceVersion=0: dial tcp 127.0.0.1:8443: getsockopt: connection refused

full log:
[log.zip](https://github.com/kubernetes/minikube/files/2513995/log.zip)

**Anything else do we need to know**:

**Comment by titanshift** [comment]
I have the exactly same problem

**Comment by tstromberg** [comment]
The issue here is that your VM isn't able to route packets to Google Public DNS (2001:4860:4860::8888) in order to download the pause-amd64:3.1 image. It doesn't help that inikube doesn't detect this condition in a user-friendly manner - yet:

```
[2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
Oct 25 08:22:31 minikube kubelet[3045]: E1025 08:22:31.587239 3045 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "kube-apiserver-minikube_kube-system(f4b05ecfb3bc2c80b4821065f19bb199)" failed: rpc error: code = Unknown desc = failed pulling image "k8s.gcr.io/pause-amd64:3.1": Error response from daemon: Get k8s.gcr.io/v2: dial tcp: lookup k8s.gcr.io on [2001:4860:4860::8888]:53: dial udp [2001:4860:4860::8888]:53: connect: network is unreachable
```

Please configure your hyperv-virtual-switch "External" appropriately. I'll leave this open as a reminder that we need to improve this situation. 

Related issues: #3031 and #2493.

**Comment by bweissman** [comment]
I am not sure the switch is the issue - other VMs in hyperv on the same switch work fine.

anyways, I worked around it by using Kubernetes from docker instead of minikube for now... will try to get back to this later.

**Comment by tstromberg** [comment]
I believe this issue was resolved in the v1.1.0 release. Please try upgrading to the latest release of minikube, and if the same issue occurs, please re-open this bug. Thank you opening this bug report, and for your patience!
